gnovm/tests/files/issue-2449.gno

```diff
@@ -0,0 +1,28 @@
+// PKGPATH: gno.land/r/evt_test
+package evt_test
+
+type Event struct {
+	name string
+}
+
+var events = []*Event{
+	{name: "event1"},
+	{name: "event2"},
+	{name: "event3"},
+	{name: "event4"},
+}
+
+func DelEvent(name string) {
+	for i, event := range events {
+		if event.name == name {
+			events = append(events[:i], events[i+1:]...)
+			return
+		}
+	}
+}
+
+func main() {
+	DelEvent("event2")
```

Comment:
Could you harden the test here by validating the content of `events` by inserting:
```go
if len(events) != 3 || events[1].name != "event3" {
   panic("test failed")
}
```

Reply:
Thanks Marc. I've added insertion and append cases for this and verify the slices' contents in the output.